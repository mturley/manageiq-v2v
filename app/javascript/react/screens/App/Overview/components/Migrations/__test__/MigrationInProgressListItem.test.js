import React from 'react';
import { shallow, mount } from 'enzyme';

import MigrationInProgressListItem from '../MigrationInProgressListItem';
import { transformationPlans } from '../../../overview.transformationPlans.fixtures';
import {
  MIGRATIONS_FILTERS,
  TRANSFORMATION_PLAN_REQUESTS_URL,
  FETCH_TRANSFORMATION_PLANS_URL
} from '../../../OverviewConstants';

describe('when the request is denied', () => {
  const plan = { ...transformationPlans.resources[7], transformation_mapping: { transformation_mapping_items: [] } };
  const [deniedRequest] = plan.miq_requests;
  const allRequestsWithTasks = [{ ...deniedRequest, miq_request_tasks: [] }];
  const baseProps = {
    plan,
    allRequestsWithTasks
  };

  test('renders an error view', () => {
    const wrapper = shallow(<MigrationInProgressListItem {...baseProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the cancel button acknowledges the dismissal and switches the view to completed migrations', async () => {
    const acknowledgeDeniedPlanRequestAction = jest.fn(() => Promise.resolve());
    const setMigrationsFilterAction = jest.fn();
    const wrapper = mount(
      <MigrationInProgressListItem
        {...baseProps}
        acknowledgeDeniedPlanRequestAction={acknowledgeDeniedPlanRequestAction}
        setMigrationsFilterAction={setMigrationsFilterAction}
      />
    );

    wrapper.find('Button').simulate('click');

    await expect(acknowledgeDeniedPlanRequestAction).toHaveBeenCalled();
    expect(setMigrationsFilterAction).toHaveBeenCalledWith(MIGRATIONS_FILTERS.completed);

    wrapper.unmount();
  });
});

describe('if there are no conversion hosts available', () => {
  const plan = { ...transformationPlans.resources[5], transformation_mapping: { transformation_mapping_items: [] } };
  const [request] = plan.miq_requests;
  const allRequestsWithTasks = [{ ...request, miq_request_tasks: [{ conversion_host_id: null }] }];
  const baseProps = {
    plan,
    allRequestsWithTasks,
    requestsProcessingCancellation: []
  };

  test('renders an error view', () => {
    const wrapper = shallow(<MigrationInProgressListItem {...baseProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the cancel button initiates cancel at the request level and refreshes plans', async () => {
    const cancelPlanRequestAction = jest.fn(() => Promise.resolve());
    const fetchTransformationPlansAction = jest.fn();
    const wrapper = mount(
      <MigrationInProgressListItem
        {...baseProps}
        cancelPlanRequestAction={cancelPlanRequestAction}
        fetchTransformationPlansAction={fetchTransformationPlansAction}
        fetchTransformationPlansUrl={FETCH_TRANSFORMATION_PLANS_URL}
      />
    );

    wrapper.find('Button').simulate('click');

    await expect(cancelPlanRequestAction).toHaveBeenCalledWith(TRANSFORMATION_PLAN_REQUESTS_URL, request.id);
    expect(fetchTransformationPlansAction).toHaveBeenCalledWith({
      url: FETCH_TRANSFORMATION_PLANS_URL,
      archived: false
    });
  });
});

describe('warm migration without cutover date set', () => {
  const plan = {
    ...transformationPlans.resources[8],
    transformation_mapping: { transformation_mapping_items: [] },
    infraMappingName: 'name'
  };
  const [request] = plan.miq_requests;
  const allRequestsWithTasks = [{ ...request, miq_request_tasks: [{ conversion_host_id: 1 }] }];
  const baseProps = {
    plan,
    allRequestsWithTasks,
    requestsProcessingCancellation: [],
    loading: false
  };

  test('renders an in progress view', () => {
    const wrapper = shallow(<MigrationInProgressListItem {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the schedule cutover button launches modal window', async () => {
    const toggleScheduleMigrationModal = jest.fn(() => Promise.resolve());
    const wrapper = mount(
      <MigrationInProgressListItem {...baseProps} toggleScheduleMigrationModal={toggleScheduleMigrationModal} />
    );

    wrapper.find('button').simulate('click');
    await expect(toggleScheduleMigrationModal).toHaveBeenCalledWith({ plan });
  });
});

describe('warm migration with cutover date set', () => {
  const plan = {
    ...transformationPlans.resources[9],
    transformation_mapping: { transformation_mapping_items: [] },
    infraMappingName: 'name'
  };
  const [request] = plan.miq_requests;
  const allRequestsWithTasks = [{ ...request, miq_request_tasks: [{ conversion_host_id: 1 }] }];
  const baseProps = {
    plan,
    allRequestsWithTasks,
    requestsProcessingCancellation: [],
    loading: false
  };

  test('renders an in progress view', () => {
    const wrapper = shallow(<MigrationInProgressListItem {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('clicking the edit cutover button launches modal window', async () => {
    const toggleScheduleMigrationModal = jest.fn(() => Promise.resolve());
    const wrapper = mount(
      <MigrationInProgressListItem {...baseProps} toggleScheduleMigrationModal={toggleScheduleMigrationModal} />
    );

    wrapper.find('a[id^="edit_cutover"]').simulate('click');
    await expect(toggleScheduleMigrationModal).toHaveBeenCalledWith({ plan });
  });

  test('clicking the delete cutover button launches confirmation window', async () => {
    const showConfirmModalAction = jest.fn(() => Promise.resolve());
    const wrapper = mount(
      <MigrationInProgressListItem {...baseProps} showConfirmModalAction={showConfirmModalAction} />
    );

    wrapper.find('a[id^="delete_cutover"]').simulate('click');
    await expect(showConfirmModalAction).toHaveBeenCalled();
  });
});
