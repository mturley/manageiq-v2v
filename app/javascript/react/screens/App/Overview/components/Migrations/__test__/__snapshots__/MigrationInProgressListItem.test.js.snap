// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if there are no conversion hosts available renders an error view 1`] = `
<InProgressRow
  actions={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Cancel Migration
    </Button>
  }
  additionalInfo={
    Array [
      <ListViewTableInfoItem
        className=""
        stacked={false}
      >
        <Spinner
          className=""
          inline={true}
          inverse={false}
          loading={true}
          size="sm"
        />
        <EllipisWithTooltip
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          Waiting for an available conversion host. You can continue waiting or go to the Migration Settings page to increase the number of migrations per host.
        </EllipisWithTooltip>
      </ListViewTableInfoItem>,
      <CutoverTimeInfoItem
        plan={
          Object {
            "configVmLength": 2,
            "created_on": "2018-05-01T12:13:50Z",
            "fulfilledOn": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_templates/60",
            "id": "60",
            "miq_requests": Array [
              Object {
                "approval_state": "approved",
                "created_on": "2018-04-06T12:31:30Z",
                "description": "Migration Plan F-0",
                "destination_id": null,
                "destination_type": null,
                "fulfilled_on": "2018-04-06T12:31:30Z",
                "href": "http://localhost:3000/api/service_requests/6000",
                "id": "6000",
                "message": "VM Transformations <SOMETHING>",
                "options": Object {
                  "cart_state": "ordered",
                  "delivered_on": "2018-04-06T12:49:30Z",
                  "dialog": null,
                  "initiator": null,
                  "requester_group": "EvmGroup-super_administrator",
                  "src_id": "60",
                  "workflow_settings": Object {
                    "resource_action_id": "2507",
                  },
                },
                "process": true,
                "request_state": "active",
                "request_type": "transformation_plan",
                "requester_id": "1",
                "requester_name": "Administrator",
                "service_order_id": "91",
                "source_id": "60",
                "source_type": "ServiceTemplate",
                "status": "Ok",
                "tenant_id": "1",
                "type": "ServiceTemplateTransformationPlanRequest",
                "updated_on": "2018-04-06T12:31:30Z",
                "userid": "admin",
              },
            ],
            "name": "Migration Plan F-0",
            "options": Object {
              "config_info": Object {
                "actions": Array [
                  Object {
                    "vm_id": "1",
                  },
                  Object {
                    "vm_id": "3",
                  },
                ],
                "transformation_mapping_id": "1",
              },
            },
            "scheduleTime": null,
            "status": "Ok",
            "transformation_mapping": Object {
              "transformation_mapping_items": Array [],
            },
          }
        }
        planRequest={
          Object {
            "approval_state": "approved",
            "created_on": "2018-04-06T12:31:30Z",
            "description": "Migration Plan F-0",
            "destination_id": null,
            "destination_type": null,
            "fulfilled_on": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_requests/6000",
            "id": "6000",
            "message": "VM Transformations <SOMETHING>",
            "miq_request_tasks": Array [
              Object {
                "conversion_host_id": null,
              },
            ],
            "options": Object {
              "cart_state": "ordered",
              "delivered_on": "2018-04-06T12:49:30Z",
              "dialog": null,
              "initiator": null,
              "requester_group": "EvmGroup-super_administrator",
              "src_id": "60",
              "workflow_settings": Object {
                "resource_action_id": "2507",
              },
            },
            "process": true,
            "request_state": "active",
            "request_type": "transformation_plan",
            "requester_id": "1",
            "requester_name": "Administrator",
            "service_order_id": "91",
            "source_id": "60",
            "source_type": "ServiceTemplate",
            "status": "Ok",
            "tenant_id": "1",
            "type": "ServiceTemplateTransformationPlanRequest",
            "updated_on": "2018-04-06T12:31:30Z",
            "userid": "admin",
          }
        }
      />,
    ]
  }
  numFailedVms={0}
  numTotalVms={0}
  onClick={null}
  plan={
    Object {
      "configVmLength": 2,
      "created_on": "2018-05-01T12:13:50Z",
      "fulfilledOn": "2018-04-06T12:31:30Z",
      "href": "http://localhost:3000/api/service_templates/60",
      "id": "60",
      "miq_requests": Array [
        Object {
          "approval_state": "approved",
          "created_on": "2018-04-06T12:31:30Z",
          "description": "Migration Plan F-0",
          "destination_id": null,
          "destination_type": null,
          "fulfilled_on": "2018-04-06T12:31:30Z",
          "href": "http://localhost:3000/api/service_requests/6000",
          "id": "6000",
          "message": "VM Transformations <SOMETHING>",
          "options": Object {
            "cart_state": "ordered",
            "delivered_on": "2018-04-06T12:49:30Z",
            "dialog": null,
            "initiator": null,
            "requester_group": "EvmGroup-super_administrator",
            "src_id": "60",
            "workflow_settings": Object {
              "resource_action_id": "2507",
            },
          },
          "process": true,
          "request_state": "active",
          "request_type": "transformation_plan",
          "requester_id": "1",
          "requester_name": "Administrator",
          "service_order_id": "91",
          "source_id": "60",
          "source_type": "ServiceTemplate",
          "status": "Ok",
          "tenant_id": "1",
          "type": "ServiceTemplateTransformationPlanRequest",
          "updated_on": "2018-04-06T12:31:30Z",
          "userid": "admin",
        },
      ],
      "name": "Migration Plan F-0",
      "options": Object {
        "config_info": Object {
          "actions": Array [
            Object {
              "vm_id": "1",
            },
            Object {
              "vm_id": "3",
            },
          ],
          "transformation_mapping_id": "1",
        },
      },
      "scheduleTime": null,
      "status": "Ok",
      "transformation_mapping": Object {
        "transformation_mapping_items": Array [],
      },
    }
  }
/>
`;

exports[`warm migration with cutover date set renders an in progress view 1`] = `
<InProgressRow
  actions={
    <div>
      <StopPropagationOnClick>
        <DropdownKebab
          className=""
          componentClass={[Function]}
          id="100-kebab"
          pullRight={true}
          toggleStyle="link"
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            id="edit_cutover_100"
            onClick={[Function]}
          >
            Edit Cutover
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            id="delete_cutover_100"
            onClick={[Function]}
          >
            Delete Scheduled Cutover
          </MenuItem>
        </DropdownKebab>
      </StopPropagationOnClick>
    </div>
  }
  additionalInfo={
    Array [
      <ListViewTableInfoItem
        className=""
        stacked={false}
      >
        <div
          className="vm-progress-bar"
          id="vm-progress-bar-100"
        >
          <UtilizationBar
            availableTooltipFunction={[Function]}
            className={null}
            description="Migrating data"
            descriptionPlacementTop={true}
            label={
              <span>
                <span
                  id="vms-migrated"
                >
                  0 of 0 VMs
                </span>
                 
                migrated
              </span>
            }
            max={0}
            min={0}
            now={0}
            thresholdError={null}
            thresholdWarning={null}
            usedTooltipFunction={[Function]}
          />
          <div
            className="info-with-inline-icon"
          >
            <Icon
              name="clock-o"
              type="fa"
            />
            <TickingIsoElapsedTime
              endTime={null}
              startTime="2018-04-06T12:31:30Z"
            />
          </div>
        </div>
      </ListViewTableInfoItem>,
      <CutoverTimeInfoItem
        plan={
          Object {
            "configVmLength": 2,
            "created_on": "2018-05-01T12:13:50Z",
            "fulfilledOn": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_templates/100",
            "id": "100",
            "infraMappingName": "name",
            "miq_requests": Array [
              Object {
                "approval_state": "approved",
                "created_on": "2018-04-06T12:31:30Z",
                "description": "Migration Plan J-0",
                "destination_id": null,
                "destination_type": null,
                "fulfilled_on": "2018-04-06T12:31:30Z",
                "href": "http://localhost:3000/api/service_requests/6000",
                "id": "6000",
                "message": "VM Transformations <SOMETHING>",
                "options": Object {
                  "cart_state": "ordered",
                  "cutover_datetime": "2030-08-14T02:34:56-06:00",
                  "delivered_on": "2018-04-06T12:49:30Z",
                  "dialog": null,
                  "initiator": null,
                  "requester_group": "EvmGroup-super_administrator",
                  "src_id": "100",
                  "workflow_settings": Object {
                    "resource_action_id": "2507",
                  },
                },
                "process": true,
                "request_state": "active",
                "request_type": "transformation_plan",
                "requester_id": "1",
                "requester_name": "Administrator",
                "service_order_id": "91",
                "source_id": "100",
                "source_type": "ServiceTemplate",
                "status": "Ok",
                "tenant_id": "1",
                "type": "ServiceTemplateTransformationPlanRequest",
                "updated_on": "2018-04-06T12:31:30Z",
                "userid": "admin",
              },
            ],
            "name": "Migration Plan J-0",
            "options": Object {
              "config_info": Object {
                "actions": Array [
                  Object {
                    "vm_id": "1",
                  },
                  Object {
                    "vm_id": "3",
                  },
                ],
                "transformation_mapping_id": "1",
                "warm_migration": true,
              },
            },
            "scheduleTime": null,
            "status": "Ok",
            "transformation_mapping": Object {
              "transformation_mapping_items": Array [],
            },
          }
        }
        planRequest={
          Object {
            "approval_state": "approved",
            "created_on": "2018-04-06T12:31:30Z",
            "description": "Migration Plan J-0",
            "destination_id": null,
            "destination_type": null,
            "fulfilled_on": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_requests/6000",
            "id": "6000",
            "message": "VM Transformations <SOMETHING>",
            "miq_request_tasks": Array [
              Object {
                "conversion_host_id": 1,
              },
            ],
            "options": Object {
              "cart_state": "ordered",
              "cutover_datetime": "2030-08-14T02:34:56-06:00",
              "delivered_on": "2018-04-06T12:49:30Z",
              "dialog": null,
              "initiator": null,
              "requester_group": "EvmGroup-super_administrator",
              "src_id": "100",
              "workflow_settings": Object {
                "resource_action_id": "2507",
              },
            },
            "process": true,
            "request_state": "active",
            "request_type": "transformation_plan",
            "requester_id": "1",
            "requester_name": "Administrator",
            "service_order_id": "91",
            "source_id": "100",
            "source_type": "ServiceTemplate",
            "status": "Ok",
            "tenant_id": "1",
            "type": "ServiceTemplateTransformationPlanRequest",
            "updated_on": "2018-04-06T12:31:30Z",
            "userid": "admin",
          }
        }
      />,
    ]
  }
  numFailedVms={0}
  numTotalVms={0}
  onClick={[Function]}
  plan={
    Object {
      "configVmLength": 2,
      "created_on": "2018-05-01T12:13:50Z",
      "fulfilledOn": "2018-04-06T12:31:30Z",
      "href": "http://localhost:3000/api/service_templates/100",
      "id": "100",
      "infraMappingName": "name",
      "miq_requests": Array [
        Object {
          "approval_state": "approved",
          "created_on": "2018-04-06T12:31:30Z",
          "description": "Migration Plan J-0",
          "destination_id": null,
          "destination_type": null,
          "fulfilled_on": "2018-04-06T12:31:30Z",
          "href": "http://localhost:3000/api/service_requests/6000",
          "id": "6000",
          "message": "VM Transformations <SOMETHING>",
          "options": Object {
            "cart_state": "ordered",
            "cutover_datetime": "2030-08-14T02:34:56-06:00",
            "delivered_on": "2018-04-06T12:49:30Z",
            "dialog": null,
            "initiator": null,
            "requester_group": "EvmGroup-super_administrator",
            "src_id": "100",
            "workflow_settings": Object {
              "resource_action_id": "2507",
            },
          },
          "process": true,
          "request_state": "active",
          "request_type": "transformation_plan",
          "requester_id": "1",
          "requester_name": "Administrator",
          "service_order_id": "91",
          "source_id": "100",
          "source_type": "ServiceTemplate",
          "status": "Ok",
          "tenant_id": "1",
          "type": "ServiceTemplateTransformationPlanRequest",
          "updated_on": "2018-04-06T12:31:30Z",
          "userid": "admin",
        },
      ],
      "name": "Migration Plan J-0",
      "options": Object {
        "config_info": Object {
          "actions": Array [
            Object {
              "vm_id": "1",
            },
            Object {
              "vm_id": "3",
            },
          ],
          "transformation_mapping_id": "1",
          "warm_migration": true,
        },
      },
      "scheduleTime": null,
      "status": "Ok",
      "transformation_mapping": Object {
        "transformation_mapping_items": Array [],
      },
    }
  }
/>
`;

exports[`warm migration without cutover date set renders an in progress view 1`] = `
<InProgressRow
  actions={
    <div>
      <ScheduleMigrationButton
        isMissingMapping={false}
        loading={false}
        plan={
          Object {
            "configVmLength": 2,
            "created_on": "2018-05-01T12:13:50Z",
            "fulfilledOn": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_templates/90",
            "id": "90",
            "infraMappingName": "name",
            "miq_requests": Array [
              Object {
                "approval_state": "approved",
                "created_on": "2018-04-06T12:31:30Z",
                "description": "Migration Plan I-0",
                "destination_id": null,
                "destination_type": null,
                "fulfilled_on": "2018-04-06T12:31:30Z",
                "href": "http://localhost:3000/api/service_requests/6000",
                "id": "6000",
                "message": "VM Transformations <SOMETHING>",
                "options": Object {
                  "cart_state": "ordered",
                  "delivered_on": "2018-04-06T12:49:30Z",
                  "dialog": null,
                  "initiator": null,
                  "requester_group": "EvmGroup-super_administrator",
                  "src_id": "90",
                  "workflow_settings": Object {
                    "resource_action_id": "2507",
                  },
                },
                "process": true,
                "request_state": "active",
                "request_type": "transformation_plan",
                "requester_id": "1",
                "requester_name": "Administrator",
                "service_order_id": "91",
                "source_id": "90",
                "source_type": "ServiceTemplate",
                "status": "Ok",
                "tenant_id": "1",
                "type": "ServiceTemplateTransformationPlanRequest",
                "updated_on": "2018-04-06T12:31:30Z",
                "userid": "admin",
              },
            ],
            "name": "Migration Plan I-0",
            "options": Object {
              "config_info": Object {
                "actions": Array [
                  Object {
                    "vm_id": "1",
                  },
                  Object {
                    "vm_id": "3",
                  },
                ],
                "transformation_mapping_id": "1",
                "warm_migration": true,
              },
            },
            "scheduleTime": null,
            "status": "Ok",
            "transformation_mapping": Object {
              "transformation_mapping_items": Array [],
            },
          }
        }
        toggleScheduleMigrationModal={undefined}
      />
    </div>
  }
  additionalInfo={
    Array [
      <ListViewTableInfoItem
        className=""
        stacked={false}
      >
        <div
          className="vm-progress-bar"
          id="vm-progress-bar-90"
        >
          <UtilizationBar
            availableTooltipFunction={[Function]}
            className={null}
            description="Migrating data"
            descriptionPlacementTop={true}
            label={
              <span>
                <span
                  id="vms-migrated"
                >
                  0 of 0 VMs
                </span>
                 
                migrated
              </span>
            }
            max={0}
            min={0}
            now={0}
            thresholdError={null}
            thresholdWarning={null}
            usedTooltipFunction={[Function]}
          />
          <div
            className="info-with-inline-icon"
          >
            <Icon
              name="clock-o"
              type="fa"
            />
            <TickingIsoElapsedTime
              endTime={null}
              startTime="2018-04-06T12:31:30Z"
            />
          </div>
        </div>
      </ListViewTableInfoItem>,
      <CutoverTimeInfoItem
        plan={
          Object {
            "configVmLength": 2,
            "created_on": "2018-05-01T12:13:50Z",
            "fulfilledOn": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_templates/90",
            "id": "90",
            "infraMappingName": "name",
            "miq_requests": Array [
              Object {
                "approval_state": "approved",
                "created_on": "2018-04-06T12:31:30Z",
                "description": "Migration Plan I-0",
                "destination_id": null,
                "destination_type": null,
                "fulfilled_on": "2018-04-06T12:31:30Z",
                "href": "http://localhost:3000/api/service_requests/6000",
                "id": "6000",
                "message": "VM Transformations <SOMETHING>",
                "options": Object {
                  "cart_state": "ordered",
                  "delivered_on": "2018-04-06T12:49:30Z",
                  "dialog": null,
                  "initiator": null,
                  "requester_group": "EvmGroup-super_administrator",
                  "src_id": "90",
                  "workflow_settings": Object {
                    "resource_action_id": "2507",
                  },
                },
                "process": true,
                "request_state": "active",
                "request_type": "transformation_plan",
                "requester_id": "1",
                "requester_name": "Administrator",
                "service_order_id": "91",
                "source_id": "90",
                "source_type": "ServiceTemplate",
                "status": "Ok",
                "tenant_id": "1",
                "type": "ServiceTemplateTransformationPlanRequest",
                "updated_on": "2018-04-06T12:31:30Z",
                "userid": "admin",
              },
            ],
            "name": "Migration Plan I-0",
            "options": Object {
              "config_info": Object {
                "actions": Array [
                  Object {
                    "vm_id": "1",
                  },
                  Object {
                    "vm_id": "3",
                  },
                ],
                "transformation_mapping_id": "1",
                "warm_migration": true,
              },
            },
            "scheduleTime": null,
            "status": "Ok",
            "transformation_mapping": Object {
              "transformation_mapping_items": Array [],
            },
          }
        }
        planRequest={
          Object {
            "approval_state": "approved",
            "created_on": "2018-04-06T12:31:30Z",
            "description": "Migration Plan I-0",
            "destination_id": null,
            "destination_type": null,
            "fulfilled_on": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_requests/6000",
            "id": "6000",
            "message": "VM Transformations <SOMETHING>",
            "miq_request_tasks": Array [
              Object {
                "conversion_host_id": 1,
              },
            ],
            "options": Object {
              "cart_state": "ordered",
              "delivered_on": "2018-04-06T12:49:30Z",
              "dialog": null,
              "initiator": null,
              "requester_group": "EvmGroup-super_administrator",
              "src_id": "90",
              "workflow_settings": Object {
                "resource_action_id": "2507",
              },
            },
            "process": true,
            "request_state": "active",
            "request_type": "transformation_plan",
            "requester_id": "1",
            "requester_name": "Administrator",
            "service_order_id": "91",
            "source_id": "90",
            "source_type": "ServiceTemplate",
            "status": "Ok",
            "tenant_id": "1",
            "type": "ServiceTemplateTransformationPlanRequest",
            "updated_on": "2018-04-06T12:31:30Z",
            "userid": "admin",
          }
        }
      />,
    ]
  }
  numFailedVms={0}
  numTotalVms={0}
  onClick={[Function]}
  plan={
    Object {
      "configVmLength": 2,
      "created_on": "2018-05-01T12:13:50Z",
      "fulfilledOn": "2018-04-06T12:31:30Z",
      "href": "http://localhost:3000/api/service_templates/90",
      "id": "90",
      "infraMappingName": "name",
      "miq_requests": Array [
        Object {
          "approval_state": "approved",
          "created_on": "2018-04-06T12:31:30Z",
          "description": "Migration Plan I-0",
          "destination_id": null,
          "destination_type": null,
          "fulfilled_on": "2018-04-06T12:31:30Z",
          "href": "http://localhost:3000/api/service_requests/6000",
          "id": "6000",
          "message": "VM Transformations <SOMETHING>",
          "options": Object {
            "cart_state": "ordered",
            "delivered_on": "2018-04-06T12:49:30Z",
            "dialog": null,
            "initiator": null,
            "requester_group": "EvmGroup-super_administrator",
            "src_id": "90",
            "workflow_settings": Object {
              "resource_action_id": "2507",
            },
          },
          "process": true,
          "request_state": "active",
          "request_type": "transformation_plan",
          "requester_id": "1",
          "requester_name": "Administrator",
          "service_order_id": "91",
          "source_id": "90",
          "source_type": "ServiceTemplate",
          "status": "Ok",
          "tenant_id": "1",
          "type": "ServiceTemplateTransformationPlanRequest",
          "updated_on": "2018-04-06T12:31:30Z",
          "userid": "admin",
        },
      ],
      "name": "Migration Plan I-0",
      "options": Object {
        "config_info": Object {
          "actions": Array [
            Object {
              "vm_id": "1",
            },
            Object {
              "vm_id": "3",
            },
          ],
          "transformation_mapping_id": "1",
          "warm_migration": true,
        },
      },
      "scheduleTime": null,
      "status": "Ok",
      "transformation_mapping": Object {
        "transformation_mapping_items": Array [],
      },
    }
  }
/>
`;

exports[`when the request is denied renders an error view 1`] = `
<InProgressRow
  actions={
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      Cancel Migration
    </Button>
  }
  additionalInfo={
    Array [
      <ListViewTableInfoItem
        className=""
        stacked={false}
      >
        <Icon
          name="error-circle-o"
          type="pf"
        />
        <EllipisWithTooltip
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          Unable to migrate VMs because no conversion host was configured at the time of the attempted migration.
           
          See the product documentation for information on configuring conversion hosts.
        </EllipisWithTooltip>
      </ListViewTableInfoItem>,
      <CutoverTimeInfoItem
        plan={
          Object {
            "configVmLength": 2,
            "created_on": "2018-05-01T12:13:50Z",
            "fulfilledOn": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_templates/80",
            "id": "80",
            "miq_requests": Array [
              Object {
                "approval_state": "denied",
                "cancelation_status": null,
                "created_on": "2018-04-06T12:31:30Z",
                "description": "Migration Plan H-0",
                "destination_id": null,
                "destination_type": null,
                "fulfilled_on": "2018-04-06T12:31:30Z",
                "href": "http://localhost:3000/api/service_requests/8000",
                "id": "8000",
                "message": "VM Transformations <SOMETHING>",
                "options": Object {
                  "cart_state": "ordered",
                  "delivered_on": "2018-04-06T12:49:30Z",
                  "dialog": null,
                  "initiator": null,
                  "requester_group": "EvmGroup-super_administrator",
                  "src_id": "60",
                  "workflow_settings": Object {
                    "resource_action_id": "2507",
                  },
                },
                "process": true,
                "request_state": "active",
                "request_type": "transformation_plan",
                "requester_id": "1",
                "requester_name": "Administrator",
                "service_order_id": "91",
                "source_id": "80",
                "source_type": "ServiceTemplate",
                "status": "Ok",
                "tenant_id": "1",
                "updated_on": "2018-04-06T12:31:30Z",
                "userid": "admin",
              },
            ],
            "name": "Migration Plan H-0",
            "options": Object {
              "config_info": Object {
                "actions": Array [
                  Object {
                    "vm_id": "1",
                  },
                  Object {
                    "vm_id": "3",
                  },
                ],
                "transformation_mapping_id": "1",
              },
            },
            "scheduleTime": null,
            "status": "Ok",
            "transformation_mapping": Object {
              "transformation_mapping_items": Array [],
            },
          }
        }
        planRequest={
          Object {
            "approval_state": "denied",
            "cancelation_status": null,
            "created_on": "2018-04-06T12:31:30Z",
            "description": "Migration Plan H-0",
            "destination_id": null,
            "destination_type": null,
            "fulfilled_on": "2018-04-06T12:31:30Z",
            "href": "http://localhost:3000/api/service_requests/8000",
            "id": "8000",
            "message": "VM Transformations <SOMETHING>",
            "miq_request_tasks": Array [],
            "options": Object {
              "cart_state": "ordered",
              "delivered_on": "2018-04-06T12:49:30Z",
              "dialog": null,
              "initiator": null,
              "requester_group": "EvmGroup-super_administrator",
              "src_id": "60",
              "workflow_settings": Object {
                "resource_action_id": "2507",
              },
            },
            "process": true,
            "request_state": "active",
            "request_type": "transformation_plan",
            "requester_id": "1",
            "requester_name": "Administrator",
            "service_order_id": "91",
            "source_id": "80",
            "source_type": "ServiceTemplate",
            "status": "Ok",
            "tenant_id": "1",
            "updated_on": "2018-04-06T12:31:30Z",
            "userid": "admin",
          }
        }
      />,
    ]
  }
  numFailedVms={0}
  numTotalVms={0}
  onClick={null}
  plan={
    Object {
      "configVmLength": 2,
      "created_on": "2018-05-01T12:13:50Z",
      "fulfilledOn": "2018-04-06T12:31:30Z",
      "href": "http://localhost:3000/api/service_templates/80",
      "id": "80",
      "miq_requests": Array [
        Object {
          "approval_state": "denied",
          "cancelation_status": null,
          "created_on": "2018-04-06T12:31:30Z",
          "description": "Migration Plan H-0",
          "destination_id": null,
          "destination_type": null,
          "fulfilled_on": "2018-04-06T12:31:30Z",
          "href": "http://localhost:3000/api/service_requests/8000",
          "id": "8000",
          "message": "VM Transformations <SOMETHING>",
          "options": Object {
            "cart_state": "ordered",
            "delivered_on": "2018-04-06T12:49:30Z",
            "dialog": null,
            "initiator": null,
            "requester_group": "EvmGroup-super_administrator",
            "src_id": "60",
            "workflow_settings": Object {
              "resource_action_id": "2507",
            },
          },
          "process": true,
          "request_state": "active",
          "request_type": "transformation_plan",
          "requester_id": "1",
          "requester_name": "Administrator",
          "service_order_id": "91",
          "source_id": "80",
          "source_type": "ServiceTemplate",
          "status": "Ok",
          "tenant_id": "1",
          "updated_on": "2018-04-06T12:31:30Z",
          "userid": "admin",
        },
      ],
      "name": "Migration Plan H-0",
      "options": Object {
        "config_info": Object {
          "actions": Array [
            Object {
              "vm_id": "1",
            },
            Object {
              "vm_id": "3",
            },
          ],
          "transformation_mapping_id": "1",
        },
      },
      "scheduleTime": null,
      "status": "Ok",
      "transformation_mapping": Object {
        "transformation_mapping_items": Array [],
      },
    }
  }
/>
`;
